src/components/services-view/services-view.controller.ts

```diff
@@ -203,6 +212,8 @@ export class ServicesViewController implements angular.IController {
 
     this.ctrl.currentFilter = category;
     this.ctrl.currentSubFilter = subCategory;
+    subCategory = category === 'all' ? null : subCategory;
+    this.$location.path(this.$location.path()).search({category: this.ctrl.currentFilter, subcategory: subCategory});
```

Comment:
We should make adding the query param doesn't cause the view to reload. I think we'll need to add `reloadOnSearch: false` here:

https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L67

Reply:
PR for this is here -> https://github.com/openshift/origin-web-console/pull/3011